codes/classical/bits/tanner.yml

```diff
@@ -10,14 +10,30 @@ logical: bits
 name: 'Tanner code'
 introduced: '\cite{doi:10.1109/TIT.1981.1056404}'
 
-description: 'Stub.'
 
-protection: ''
+description: 'Let G(V,E), \(|V| = n \), be a \(\Delta\)-regular (not necessarily bipartite) graph and let \(C_0[\Delta,\Delta R_0] \subseteq \mathbb{F}_2^N\) be a binary linear code of rate \(R\), with N given be \(N = |E|= n\Delta/2\). The coordinates of \(C_0\) are in 1 to 1 correspondence with E. The Tanner code \(X(G,C_0)\) is defined as the set \(\{c\in \mathbb{F}_2^{n}| \forall v \in V, c_{|N(v)}\in C_0\}\). Dimension of C is \(\geq N -n(\Delta -\Delta R_0) = N(2R_0-1)\geq 0\), if \(R_0 \geq \frac{1}{2}\). Expansion properties of the underlying graphs give efficient decoding algorithms.'
+
+protection: 'Tanner Codes protect against noise on classical bit strings. If \(C_0\) is an \([d, d-t,d''> d(\gamma_0 +\frac{\lambda}{d})]_2\) code and G is an \((N, M, 2, d, \rho,\alpha)\)- expander where \(\rho = \gamma_0 (\gamma_0 +\frac{\lambda}{d})\), then the Tanner Code \(T(G, C_0)\) has rate \(1-\frac{M}{N}t\) and relative distance \(\geq \gamma_0(\gamma_0+\frac{\lambda}{d})\)'
+
+features:
+
+  rate: 'For \(C_0\) with rate R, the Tanner code has rate at least 2R-1. For \(C_0\) satisfying Gilbert-Varshamov bound, the rate vs relative distance for the Tanner code is given by \(R \geq (1-2h(\delta_0))\), where \(\delta_0\) is the relative distance of \(C_0\).'
+
+  encoders:
+    - 'Quadratic algorithm: This technique works for all linear block codes and encodes using matrix multiplication \cite{doi:10.1145/258533.258575}.'
+    - 'Using non-abelian Fast Fourier Transform and exploiting the symmetry of underlying group G, an encoding algorithm that requires \(O(n^{4/3})\) has been devised in \cite{doi:10.1145/258533.258575}.'
+    - 'Spielman modifies the code construction to obtain codes that may be encoded in linear time but give similar performance \cite{10.1109/18.556668}.'
+
+  decoders:
+    - 'Parallel decoding algorithm \cite{doi:10.1109/18.556667} corrects a fraction \(\delta_0^2/48\) of errors.'
+    - 'Decoding algorithm \cite{doi:10.1109/18.910593} gives improved result of correcting a fraction of \(\delta_0^2/4\) of errors for Tanner codes.'
+  
 
 notes:
-  - ''
 
 relations:
   parents:
+    - code_id: expander
```

Comment:
@AlexanderBarg Putting tanner as parent to expander (as opposed to the other way).